### PR TITLE
Made better address validation

### PR DESCRIPTION
**Pull request by blurpesec**
### Description
Made better address validation? I hope

#### ToDos:
- [x] Unit tests m'dude.

### CODE REVIEW

File: common/v2/services/EthService/validators.ts [hunk 1]
```diff
@@ -53,18 +56,49 @@ function getIsValidAddressFunction(chainId: number) {
   return isValidETHAddress;
 }
 
-function isValidETHLikeAddress(address: string, extraChecks?: () => boolean): boolean {
+export function isValidETHLikeAddress(address: string, extraChecks?: () => boolean): boolean {
   if (address.substring(0, 2) !== '0x') {
+    console.debug('[isValidETHLikeAddress]: return 1');
```
Comment (vilsbole) [inline common/v2/services/EthService/validators.ts]: shall we remove these logs?

File: common/v2/services/EthService/validators.ts [hunk 2]
```diff
@@ -53,18 +56,49 @@ function getIsValidAddressFunction(chainId: number) {
...
   } else if (/^(0x)?[0-9a-f]{40}$/.test(address) || /^(0x)?[0-9A-F]{40}$/.test(address)) {
+    console.debug('[isValidETHLikeAddress]: return 3');
     return true;
   } else {
+    console.debug('[isValidETHLikeAddress]: return 4');
     return extraChecks ? extraChecks() : false;
   }
 }
 
+export const isValidETHRecipientAddress = (
```
Comment (vilsbole) [inline common/v2/services/EthService/validators.ts]: Unit test please.

File: common/v2/services/EthService/validators.ts [hunk 3]
```diff
@@ -53,18 +56,49 @@ function getIsValidAddressFunction(chainId: number) {
...
+  if (isValidENSName(address) && !resolutionErr) {
+    return {
+      success: false,
+      name: 'ValidationError',
+      type: InlineMessageType.INFO_CIRCLE,
+      message: translate('TO_FIELD_ERROR')
+    };
+  }
+  if (!isValidETHAddress(address) && !isChecksumAddress(address)) {
+    if (/^(0x|0X[a-f0-9]{40}$|0x|0X[A-F0-9]{40}$)/.test(address)) {
```
Comment (vilsbole) [inline common/v2/services/EthService/validators.ts]: 1. Please refactor to remove nested conditional.
2. What is the difference between `isValidETHAddress` and the regex?.

**Comment by FrederikBolding** [comment]
[ch5306]

File: common/v2/services/EthService/__tests__/validators.spec.ts [hunk 4]
```diff
@@ -25,3 +26,68 @@ describe('isTransactionFeeHigh', () => {
...
+  it('returns false for an invalid ens name', () => {
+    const expected = { success: false };
+    const testAddress = 'mycryptoid.ethhhhh';
+    const returned = isValidETHRecipientAddress(
+      testAddress,
+      ('Domain mycryptoid.ethhhh is not supported' as unknown) as ResolutionError
+    );
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns false for an unresolved ens name', () => {
```
Comment (409H) [inline common/v2/services/EthService/__tests__/validators.spec.ts]: What stops someone from "grief attacking" us by registering this ENS name to slow down a release - maybe even if its a hotfix release? or am I too paranoid?

Also the `testAddress` and the `returned` string have different ENS names in it - not sure if that's a _problem_.

Comment (blurpesec) [reply common/v2/services/EthService/__tests__/validators.spec.ts]: woopsies

**Pull request merged**